app/webpacker/scripts/global/disable-browser-autofill.js

```diff
@@ -0,0 +1,23 @@
+function disableBrowserAutofill () {
+  const dataAttributeName = 'data-nameoriginal'
+  const inputTypes = ['text', 'tel', 'email', 'number']
+
+  document.querySelectorAll('form').forEach(form => {
+    if (form.getAttribute('autocomplete') === 'off') {
+      const inputs = [...form.querySelectorAll('input[autocomplete]')].filter(e => inputTypes.includes(e.type))
+
+      inputs.forEach(input => {
+        input.setAttribute(dataAttributeName, input.name)
+        input.name = Math.random().toString(36).substring(7) // NOSONAR
+      })
+
+      form.addEventListener('submit', () => {
+        inputs.forEach(input => {
+          input.name = input.getAttribute(dataAttributeName)
+        })
+      })
```

Comment:
I wonder if you just assign a random number at the end of input.name instead of changing it completely, and regex-removing it in the backend. This way you don't have to store the original names in memory and rely on them later. 
```suggestion
      inputs.forEach(input => {
        input.name = `${input.name}-AF-Math.random().toString(36).substring(7)` // NOSONAR
      })
```

and actually I'm thinking if you append the random value via your backend to bypass the need for using client side js.

Reply:
The team had discussed this idea, but we felt it was way too much work for the returned value. It would be a Rails anti-pattern and could introduce more bugs.

Reply:
@jenbutongit we discussed this - but that's a much bigger architectural change. Whereas this is super lightweight.

Splitting the difference, the temp name could have the random number prepended  / appended to the existing - and then removed as current.